keel-actuator/src/test/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuatorTests.kt

```diff
@@ -276,6 +281,36 @@ internal class ResourceActuatorTests : JUnit5Minutests {
             verify { publisher.publishEvent(ResourceChecked(resource, Error)) }
           }
         }
+
+        context("the app badapp is denied by the policy") {
+          val badresource = Resource(
+            apiVersion = SPINNAKER_API_V1.subApi("plugin1"),
+            kind = "foo",
+            metadata = mapOf(
+              "name" to "badapp-resource",
+              "uid" to randomUID(),
+              "serviceAccount" to "keel@spinnaker"
+            ),
+            spec = DummyResourceSpec("badapp")
+          )
+          before {
+            resourceRepository.store(badresource)
+            resourceRepository.appendHistory(ResourceCreated(badresource))
+            coEvery { plugin1.desired(badresource) } returns DummyResource(badresource.spec.state)
+            coEvery { plugin1.current(badresource) } returns DummyResource(badresource.spec.state)
+            coEvery { plugin1.update(badresource, any()) } returns listOf(TaskRef("/tasks/${randomUID()}"))
```

Comment:
Why do you need to stub these if it's not going to check that resource? I would have thought it should not call them (might make a good additional test).

Reply:
Did this before I figured out what test I actually wanted to create. Good catch!